backend/plugin/db/mongodb/mongodb.go

```diff
@@ -173,8 +173,8 @@ func getMongoDBConnectionURI(connConfig db.ConnectionConfig) string {
 		connectionURI = "mongodb+srv://"
 	}
 	if connConfig.Username != "" {
-		percentEncodingUsername := replaceCharacterWithPercentEncoding(connConfig.Username)
-		percentEncodingPassword := replaceCharacterWithPercentEncoding(connConfig.Password)
+		percentEncodingUsername := url.QueryEscape(connConfig.Username)
```

Comment:
Cool! Is it possible to use url library for the whole thing?

Reply:
Done.